src/wp-includes/comment-template.php

```diff
@@ -1969,16 +1973,20 @@ function cancel_comment_reply_link( $text = '' ) {
  * Retrieves hidden input HTML for replying to comments.
  *
  * @since 3.0.0
+ * @since 6.2.0 Renamed `$post_id` to `$post` and added WP_Post support.
  *
- * @param int $post_id Optional. Post ID. Defaults to the current post ID.
+ * @param int|WP_Post $post Optional. The post the comment is being displayed for.
```

Comment:
Also needs to `null` for the `@param`. Why? 

- `null` is the default
- it is acceptable to pass `null`
- for parity with `get_post()` which documents this param as `int|WP_Post|null`